### PR TITLE
feat(rubygems): API throttling for `rubygems.org`

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

According to official docs, `rubygems.org` host is rate-limited: https://guides.rubygems.org/rubygems-org-rate-limits/

This PR enforces this limits.

## Context

- Ref: #3670

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/rubygems/http.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Http } from '../../../util/http';
+import { Throttle } from '../../../util/http/throttle';
+import { parseUrl } from '../../../util/url';
+
+export class RubygemsHttp extends Http {
+  protected override getThrottle(url: string): Throttle | null {
+    const host = parseUrl(url)?.host;
+
+    if (host === 'rubygems.org') {
+      const intervalMs = 100;
```
Comment (rarkins) [inline lib/modules/datasource/rubygems/http.ts:10]:
```suggestion
      // rubygems.org has a rate limit of 15 per second, so we use a more conservative 10
      const intervalMs = 100;
```

Comment (zharinov) [reply lib/modules/datasource/rubygems/http.ts:10]: Actually, I think we're not conservative:
- API and website: 10 requests per second
- Dependency API: 15 requests per second

Comment (zharinov) [reply lib/modules/datasource/rubygems/http.ts:10]: (dependency API is the deprecated one, so we're referring to first one)

File: lib/modules/datasource/rubygems/http.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Http } from '../../../util/http';
+import { Throttle } from '../../../util/http/throttle';
+import { parseUrl } from '../../../util/url';
+
+export class RubygemsHttp extends Http {
+  protected override getThrottle(url: string): Throttle | null {
+    const host = parseUrl(url)?.host;
+
+    if (host === 'rubygems.org') {
+      // rubygems.org has a rate limit of 10 per second, so we use a more conservative 8
```
Comment (rarkins) [inline lib/modules/datasource/rubygems/http.ts:10]: Are you sure it's the 10/s which applies and not 15/s?

**Review by viceice — APPROVED** [review]

File: lib/modules/datasource/rubygems/http.ts (hunk 2 above)
Comment (rarkins) [reply lib/modules/datasource/rubygems/http.ts:10]: But I'm ok with this actually, considering how many workers we may have in parallel

Comment (zharinov) [reply lib/modules/datasource/rubygems/http.ts:10]: According to [this page](https://blog.rubygems.org/2023/02/22/dependency-api-deprecation.html), dependencies API is the one which we use as fallback:

> The API has a single endpoint, `/api/v1/dependencies` (returning responses as Marshal or as JSON), accepting a single comma-separated gems query parameter.

It means, that other endpoints are in "API and website" category, which is 10 per second.

**Pull request merged**